src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -190,6 +190,10 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
     // bottom in the new buffer as well. Track that case now.
     const bool originalOffsetWasZero = _scrollOffset == 0;
 
+    // skip any drawing updates that might occur until we swap _buffer with the new buffer or if we exit early.
+    _buffer->GetCursor().StartDeferDrawing();
+    auto endDefer = wil::scope_exit([&]() noexcept { _buffer->GetCursor().EndDeferDrawing(); });

```

Comment:
nit: might be worth a comment that we're capturing `_buffer` _by reference_ because either way, when we exit, we want to undefer the _current active_ one